### ISSUE TITLE
object_detection: Continuous evaluation in model_main picks up the wrong checkpoint-dir

### ISSUE THREAD

**Issue by volkerstampa** [tensorflow/models#5208]
### System information
- **What is the top-level directory of the model you are using**: object_detection
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: no
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04.5 LTS (Xenial Xerus)
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: ('v1.10.1-0-g4dcfddc5d1', '1.10.1')
- **Bazel version (if compiling from source)**: -
- **CUDA/cuDNN version**: cuda-9.0
- **GPU model and memory**:  Tesla K80 major: 3 minor: 7 / 11.17GiB 
- **Exact command to reproduce**:
  ```
  python -m object_detection.model_main \
      --logtostderr \
      --pipeline_config_path "models/dev_pipeline.config" \
      --model_dir "models/dev" \
      --checkpoint_dir "models/train/" 
  ```
### Describe the problem
I am trying to start a continuous evaluation with `model_main.py` in a similar way as it could be started with `eval.py` (that was moved the the legacy folder). By providing the _train_-`model_dir` as `checkpoint_dir` it starts in evaluation mode as expected. Without `--run_once` it start continuous evaluation with:
```
      model_lib.continuous_eval(estimator, FLAGS.model_dir, input_fn,
                                eval_steps, train_steps, name)
```
This makes it use the _dev_-`model_dir` as `checkpoint_dir`, i.e. it waits for new checkpoints in `models/dev`. I believe this should read:
```
      model_lib.continuous_eval(estimator, FLAGS.checkpoint_dir, input_fn,
                                eval_steps, train_steps, name)
```
So that it uses the _train_-`model_dir` (`models/train`) as `checkpoint_dir` as this is where new checkpoints are created during training.

### Source code / logs
See above.

**Comment by pkulzc** [comment]
Yeah thanks for finding this. You could either send a PR to fix this or wait until we fix it in next release.

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.

**Comment by volkerstampa** [comment]
Fixed by #5289
